argilla-server/src/argilla_server/validators/records.py

```diff
@@ -108,6 +109,36 @@ def _validate_image_field(self, field_name: str, field_value: Union[str, None])
         else:
             raise UnprocessableEntityError(f"image field {field_name!r} has an invalid URL value")
 
+    def _validate_chat_fields(self, dataset: Dataset, fields: Dict[str, str]) -> None:
+        for field in filter(lambda field: field.is_chat, dataset.fields):
+            self._validate_chat_field(field.name, fields.get(field.name))
+
+    def _validate_chat_field(self, field_name: str, field_value: Union[str, None]) -> None:
+        if field_value is None:
+            return
+
+        if len(field_value) > 5000:
+            raise UnprocessableEntityError(
+                f"chat field {field_name!r} value is exceeding the maximum length of 5000 characters"
+            )
+
+        if not isinstance(field_value, list):
+            raise UnprocessableEntityError(f"chat field {field_name!r} value must be a list of dictionaries")
+
+        for i, value in enumerate(field_value):
+            if not isinstance(value, dict):
+                raise UnprocessableEntityError(
+                    f"chat field {field_name!r} value must be a list of dictionaries. Found a non-dictionary value at index {i}"
+                )
+            if "content" not in value:
+                raise UnprocessableEntityError(
+                    f"chat field {field_name!r} value must be a list of dictionaries with a 'content' key. Missing 'content' key at index {i}"
+                )
+            if "role" not in value:
+                raise UnprocessableEntityError(
+                    f"chat field {field_name!r} value must be a list of dictionaries with a 'role' key. Missing 'role' key at index {i}"
+                )
```

Comment:
Similar to the mention earlier about `index` not always being helpful directly. Perhaps we can show the record or the value?